### DISCUSSION TITLE
version range

### DISCUSSION THREAD

**Discussion by etylsarin** [jscutlery/semver#744] [Q&A] [answered]
Hi guys,
Is it possible to set a semver range prefix when updating dependencies in package.json files?
to generate `"my-lib": "^1.2.3"` instead of `"my-lib": "1.2.3"`

**Comment by edbzn** [comment]
Hi @etylsarin. It's on the consumer side to set the range, not on the package itself, right? So we can't act for consumers, they choose whatever range they want and we can only publish a specific version. Let me know if I misunderstood your question.

**Comment by etylsarin** [comment]
Hi @edbzn thanks for the reply and sorry for confusion. I thought that what does the `trackDeps` flag do is that it bumps version of an updated package in dependencies section of `package.json` file of a consumer package. But it only bumps a version of the consumer package itself, leaving the dependencies untouched, right? And why is that flag needed when NX has the `affected` so you can run the semver task on all affected packages?

**Reply by edbzn** [reply, marked as answer]
>  But it only bumps a version of the consumer package itself, leaving the dependencies untouched, right? 

Yes exactly.

> And why is that flag needed when NX has the affected so you can run the semver task on all affected packages?

Technically @jscutlery/semver runs on project's level, it doesn't know if you are using affected or directly running the target. The `--trackDeps` flag changes the behavior of the version calculation and was introduced later in the plugin development, to avoid breaking users the flag is disabled.
